pennylane_lightning/lightning_qubit.py

```diff
@@ -964,6 +964,16 @@ def var(self, observable, shot_range=None, bin_size=None):
 
         return M.var(observable.name, observable_wires)
 
+    def _get_diagonalizing_gates(self, circuit: qml.tape.QuantumTape) -> List[Operation]:
+        meas_filtered = [
+            m
+            for m in circuit.measurements
+            if m.obs is None
+            or not isinstance(m.obs, qml.Hamiltonian)
+            and (not isinstance(m.obs, qml.ops.Sum) or m.obs._pauli_rep is None)
```

Comment:
Can this logical statement be tidied up a bit? I think there are could be a better way to express the same intent.
Also, worth noting the order here, and relying on the most likely to fail candidate up-front to avoid evaluating the other terms, and replacing up-front `or` with `and` if it makes sense to do so. Maybe applying de-Morgan's rules here can help reshape it as such.

Also, this still depends on that private attribute. Same as above, if this can be handled another way we should.

Reply:
Yea, I think having a lambda, or even a separate function could be a good call here.